lib/puppet/util/at_fork.rb

```diff
@@ -13,7 +13,7 @@
 #   service.
 module Puppet::Util::AtFork
   @handler_class = loop do
-    if Facter.value(:operatingsystem) == 'Solaris'
+    if Puppet::Util::Platform.solaris?
```

Comment:
Could you pull the at_fork and platform change for solaris into a separate commit?